# Computational genomics papers

## Quality control and batch effects

 - AC't Hoen, Peter, et al. "[Reproducibility of high-throughput mRNA and small RNA sequencing across laboratories](http://dx.doi.org/10.1038/nbt.2702)." _Nature biotechnology_ 31.11 (2013): 1015.
 - Leek, Jeffrey T., et al. "[Tackling the widespread and critical impact of batch effects in high-throughput data.](http://dx.doi.org/10.1038/nrg2825)" _Nature Reviews Genetics_ 11.10 (2010): 733.
 - Fort, Alexandre, et al. "[MBV: a method to solve sample mislabeling and detect technical bias in large combined genotype and sequencing assay datasets.](https://doi.org/10.1093/bioinformatics/btx074)" _Bioinformatics_33.12 (2017): 1895-1897.
> Detect mislabeled individuals between VCF and BAM files.

## Allelic mapping bias

 - Panousis, Nikolaos I., et al. "[Allelic mapping bias in RNA-sequencing is not a major confounder in eQTL studies.](https://doi.org/10.1186/s13059-014-0467-2)" _Genome biology_ 15.9 (2014): 467.
 - Castel, Stephane E., et al. "[Tools and best practices for data processing in allelic expression analysis.](http://dx.doi.org/10.1186/s13059-015-0762-6)" _Genome biology_16.1 (2015): 195.

## Data normalisation

 - Wagner, GÃ¼nter P., Koryu Kin, and Vincent J. Lynch. "[Measurement of mRNA abundance using RNA-seq data: RPKM measure is inconsistent among samples](https://doi.org/10.1007/s12064-012-0162-3)." _Theory in biosciences_ 131.4 (2012): 281-285. 
 
 > Formal explanation of why **T**ranscripts **P**er **M**illion
> (**TPM**) is preferable to **R**eads **P**er **K**ilobase per
> **M**illion of mapped reads (**RPKM**) when normalising sequencing read counts. See also [this blog
> post](https://www.rna-seqblog.com/rpkm-fpkm-and-tpm-clearly-explained/)
> for less formal explanation

.

## Regulatory genetic variation

 - Albert, Frank W., and Leonid Kruglyak. "[The role of regulatory variation in complex traits and disease.](http://dx.doi.org/10.1038/nrg3891)" _Nature Reviews Genetics_ 16.4 (2015): 197. 
 - Park, Eddie, et al. "[The expanding landscape of alternative splicing variation in human populations.](http://dx.doi.org/10.1016/j.ajhg.2017.11.002)" _The American Journal of Human Genetics_ 102.1 (2018): 11-26.

## Chromatin accessibility

 - Degner, Jacob F., et al. "[DNase I sensitivity QTLs are a major determinant of human expression variation](https://doi.org/10.1038/nature10808)." _Nature_ 482.7385 (2012): 390.
 - Buenrostro, Jason D., et al. "[Transposition of native chromatin for fast and sensitive epigenomic profiling of open chromatin, DNA-binding proteins and nucleosome position.](https://doi.org/10.1038/nmeth.2688)"  _Nature methods_  10.12 (2013): 1213.
> The original ATAC-seq paper.

## General advice

 - Mensh, Brett, and Konrad Kording. "[Ten simple rules for structuring papers](https://doi.org/10.1371/journal.pcbi.1005830)." _PLoS computational biology_ 13.9 (2017): e1005619.
 - Wilson, Greg, et al. "[Good enough practices in scientific computing.](http://dx.doi.org/10.1371/journal.pcbi.1005510)" _PLoS computational biology_ 13.6 (2017): e1005510.
 - Wilson, Greg, et al. "[Best practices for scientific computing.](http://dx.doi.org/10.1371/journal.pbio.1001745)" _PLoS biology_ 12.1 (2014): e1001745.
 - Noble, William Stafford. "[A quick guide to organizing computational biology projects.](https://doi.org/10.1371/journal.pcbi.1000424)" _PLoS computational biology_5.7 (2009): e1000424.

